src/scene/vcFlythrough.cpp

```diff
@@ -432,3 +432,38 @@ void vcFlythrough::UpdateLinePoints()
     udFree(pPositions);
   }
 }
+
+void vcFlythrough::SmoothFlightPoints()
+{
+  size_t numCulledFPs = 0;
+  size_t flightPointIndex = 1;
+  while (flightPointIndex < m_flightPoints.length - 1)
```

Comment:
Just decrement `flightPointIndex` when you remove an item from the array instead of nesting loops.